apollo-router-core/src/spec/fragments.rs

```diff
@@ -23,9 +23,27 @@ impl From<&ast::Document> for Fragments {
                         .to_string();
                     let selection_set = fragment_definition
                         .selection_set()
-                        .expect("the node SelectionSet is not optional in the spec; qed");
+                        .expect("the node SelectionSet is not optional in the spec; qed")
+                        .selections()
+                        .map(Into::into)
+                        .collect();
+                    let type_condition = fragment_definition
+                        .type_condition()
+                        .expect("Fragments must specify the type they apply to; qed")
+                        .named_type()
+                        .expect("Fragments must specify the type they apply to; qed")
+                        .name()
+                        .expect("the node Name is not optional in the spec; qed")
+                        .text()
+                        .to_string();
```

Comment:
Maybe create a function to get "type_condition" since you do the same thing in selection.rs? Also: maybe vary the strings for the two type expects?